manager/network/model.go

```diff
@@ -0,0 +1,15 @@
+package network
+
+import (
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/allocator/networkallocator"
+)
+
+// Model is an abstraction over the Network Model to be used.
+type Model interface {
+	Allocator() (networkallocator.NetworkAllocator, error)
+	SetDefaults(spec *api.NetworkSpec) error
+	ValidateNetworkSpec(spec *api.NetworkSpec) error
+	PredefinedNetworks() []networkallocator.PredefinedNetworkData
```

Comment:
I feel like the definition of `PredefinedNetworkData` belongs in this package rather than `networkallocator`.

We can't move it on its own because that would cause a circular dependency.

I wonder if it would also make sense to move the `NetworkAllocator` interface definition here. Then this package would contain the interface definitions, and `networkallocator` would have the implementations.

I'm not completely sure either way, just throwing this out as an idea. Something doesn't feel quite right about this package depending on the allocator.